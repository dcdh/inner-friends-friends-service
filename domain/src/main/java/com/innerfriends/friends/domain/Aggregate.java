package com.innerfriends.friends.domain;

import java.util.Objects;
import java.util.concurrent.Callable;

public abstract class Aggregate {

    private Version version;

    public Aggregate(final Version version) {
        this.version = version;
    }

    protected void apply(final Callable apply) {
        try {
            apply.call();
        } catch (final Exception exception) {
            throw new FailedToMutateAggregateException(exception);
        }
        this.version = this.version.increment();
    }

    public Version version() {
        return version;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (!(o instanceof Aggregate)) return false;
        final Aggregate aggregate = (Aggregate) o;
        return Objects.equals(version, aggregate.version);
    }

    @Override
    public int hashCode() {
        return Objects.hash(version);
    }

    @Override
    public String toString() {
        return "Aggregate{" +
                "version=" + version +
                '}';
    }
}
