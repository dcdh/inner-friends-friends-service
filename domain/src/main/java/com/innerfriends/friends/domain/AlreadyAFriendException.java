package com.innerfriends.friends.domain;

public class AlreadyAFriendException extends RuntimeException {
}
