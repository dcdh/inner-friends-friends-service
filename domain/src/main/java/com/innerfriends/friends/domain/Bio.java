package com.innerfriends.friends.domain;

import java.util.Objects;

public final class Bio {

    private final String content;

    public Bio() {
        this("");
    }

    public Bio(final String content) {
        this.content = Objects.requireNonNull(content);
        if (content.length() >= 500) {
            throw new IllegalStateException();
        }
    }

    public String content() {
        return content;
    }

    public boolean hasBio() {
        return content.length() > 0;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (!(o instanceof Bio)) return false;
        final Bio bio1 = (Bio) o;
        return Objects.equals(content, bio1.content);
    }

    @Override
    public int hashCode() {
        return Objects.hash(content);
    }
}
