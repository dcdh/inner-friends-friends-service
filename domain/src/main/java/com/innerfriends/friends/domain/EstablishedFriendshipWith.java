package com.innerfriends.friends.domain;

import java.util.Objects;

public final class EstablishedFriendshipWith {

    private final InFriendshipWithId inFriendshipWithId;

    public EstablishedFriendshipWith(final InFriendshipWithId inFriendshipWithId) {
        this.inFriendshipWithId = Objects.requireNonNull(inFriendshipWithId);
    }

    public InFriendshipWithId inFriendshipWithId() {
        return inFriendshipWithId;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (!(o instanceof EstablishedFriendshipWith)) return false;
        final EstablishedFriendshipWith that = (EstablishedFriendshipWith) o;
        return Objects.equals(inFriendshipWithId, that.inFriendshipWithId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(inFriendshipWithId);
    }

    @Override
    public String toString() {
        return "EstablishedFriendshipWith{" +
                "inFriendshipWithId=" + inFriendshipWithId +
                '}';
    }

}
