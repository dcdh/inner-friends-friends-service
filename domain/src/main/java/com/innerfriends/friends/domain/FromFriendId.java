package com.innerfriends.friends.domain;

import java.util.Objects;

public final class FromFriendId implements Pseudo {

    private final PseudoId pseudoId;

    public FromFriendId(final PseudoId pseudoId) {
        this.pseudoId = Objects.requireNonNull(pseudoId);
    }

    public FromFriendId(final String pseudo) {
        this(new PseudoId(pseudo));
    }

    @Override
    public PseudoId pseudoId() {
        return pseudoId;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (!(o instanceof FromFriendId)) return false;
        final FromFriendId that = (FromFriendId) o;
        return Objects.equals(pseudoId, that.pseudoId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(pseudoId);
    }

    @Override
    public String toString() {
        return "FromFriendId{" +
                "pseudoId=" + pseudoId +
                '}';
    }
}
