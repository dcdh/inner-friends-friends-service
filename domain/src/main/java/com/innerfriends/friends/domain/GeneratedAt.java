package com.innerfriends.friends.domain;

import java.time.LocalDateTime;
import java.util.Objects;

public final class GeneratedAt {

    private final LocalDateTime at;

    public GeneratedAt(final LocalDateTime at) {
        this.at = Objects.requireNonNull(at);
    }

    public LocalDateTime at() {
        return at;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (!(o instanceof GeneratedAt)) return false;
        final GeneratedAt that = (GeneratedAt) o;
        return Objects.equals(at, that.at);
    }

    @Override
    public int hashCode() {
        return Objects.hash(at);
    }

    @Override
    public String toString() {
        return "GeneratedAt{" +
                "at=" + at +
                '}';
    }
}
