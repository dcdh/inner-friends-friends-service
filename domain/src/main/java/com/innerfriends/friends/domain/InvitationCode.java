package com.innerfriends.friends.domain;

import java.util.Objects;
import java.util.UUID;

public final class InvitationCode {

    private final UUID code;

    public InvitationCode(final UUID code) {
        this.code = Objects.requireNonNull(code);
    }

    public UUID code() {
        return code;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (!(o instanceof InvitationCode)) return false;
        final InvitationCode that = (InvitationCode) o;
        return Objects.equals(code, that.code);
    }

    @Override
    public int hashCode() {
        return Objects.hash(code);
    }

    @Override
    public String toString() {
        return "InvitationCode{" +
                "code=" + code +
                '}';
    }
}
