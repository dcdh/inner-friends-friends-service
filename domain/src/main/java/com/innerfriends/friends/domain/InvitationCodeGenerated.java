package com.innerfriends.friends.domain;

import java.util.Objects;

public final class InvitationCodeGenerated {

    private final FromFriendId fromFriendId;
    private final InvitationCode invitationCode;
    private final GeneratedAt generatedAt;

    public InvitationCodeGenerated(final FromFriendId fromFriendId,
                                   final InvitationCode invitationCode,
                                   final GeneratedAt generatedAt) {
        this.fromFriendId = Objects.requireNonNull(fromFriendId);
        this.invitationCode = Objects.requireNonNull(invitationCode);
        this.generatedAt = Objects.requireNonNull(generatedAt);
    }

    public FromFriendId fromFriendId() {
        return fromFriendId;
    }

    public InvitationCode invitationCode() {
        return invitationCode;
    }

    public GeneratedAt generatedAt() {
        return generatedAt;
    }

    public final Boolean isValid(final LocalDateTimeProvider localDateTimeProvider) {
        return localDateTimeProvider.now().isBefore(generatedAt.at().plusMinutes(15l));
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (!(o instanceof InvitationCodeGenerated)) return false;
        final InvitationCodeGenerated that = (InvitationCodeGenerated) o;
        return Objects.equals(fromFriendId, that.fromFriendId) &&
                Objects.equals(invitationCode, that.invitationCode) &&
                Objects.equals(generatedAt, that.generatedAt);
    }

    @Override
    public int hashCode() {
        return Objects.hash(fromFriendId, invitationCode, generatedAt);
    }

    @Override
    public String toString() {
        return "InvitationCodeGenerated{" +
                "fromFriendId=" + fromFriendId +
                ", invitationCode=" + invitationCode +
                ", generatedAt=" + generatedAt +
                '}';
    }
}
