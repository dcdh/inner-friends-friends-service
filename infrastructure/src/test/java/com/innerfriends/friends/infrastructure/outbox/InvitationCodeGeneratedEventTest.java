package com.innerfriends.friends.infrastructure.outbox;

import com.innerfriends.friends.domain.FromFriendId;
import com.innerfriends.friends.domain.GeneratedAt;
import com.innerfriends.friends.domain.InvitationCode;
import com.innerfriends.friends.domain.InvitationCodeGenerated;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.UUID;

import static org.approvaltests.Approvals.verifyJson;
import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.doReturn;

@ExtendWith(MockitoExtension.class)
public class InvitationCodeGeneratedEventTest extends EventTest {

    @Test
    public void should_return_expected_event() {
        // Given
        final InvitationCodeGenerated invitationCodeGenerated = new InvitationCodeGenerated(new FromFriendId("Mario"),
                new InvitationCode(new UUID(0, 0)),
                new GeneratedAt(LocalDateTime.of(2021, 11, 10, 0, 0 ,0, 0)));
        doReturn(Instant.ofEpochSecond(1)).when(instantProvider).now();

        // When
        final InvitationCodeGeneratedEvent invitationCodeGeneratedEvent = InvitationCodeGeneratedEvent.of(invitationCodeGenerated, objectMapper, instantProvider);

        // Then
        assertThat(invitationCodeGeneratedEvent.getAggregateId()).isEqualTo("Mario");
        assertThat(invitationCodeGeneratedEvent.getAggregateType()).isEqualTo("Friend");
        assertThat(invitationCodeGeneratedEvent.getTimestamp()).isEqualTo(Instant.ofEpochSecond(1));
        assertThat(invitationCodeGeneratedEvent.getType()).isEqualTo("InvitationCodeGenerated");
        verifyJson(invitationCodeGeneratedEvent.getPayload().toString());
    }
}
