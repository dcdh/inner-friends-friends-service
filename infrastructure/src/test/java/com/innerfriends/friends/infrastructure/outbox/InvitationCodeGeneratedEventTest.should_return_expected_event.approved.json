{
  "fromFriendId": "Mario",
  "invitationCode": "00000000-0000-0000-0000-000000000000",
  "generatedAt": "2021-11-10T00:00:00"
}